zoo/board_games/mcts_bot.py

```diff
@@ -125,17 +159,35 @@ def n(self):
         return self._number_of_visits
 
     def expand(self):
+        # Choose an untried action in the list and pop it out. Only untried actions are left in the lsit.
         action = self.legal_actions.pop()
+        # simulate_action() returns a new environment object which reset the board and the current player flag.
         next_simulator_env = self.env.simulate_action(action)
+        # Create a new node object for the child node.
         child_node = TwoPlayersMCTSNode(next_simulator_env, parent=self)
         self.children.append(child_node)
         self.parent_action.append(action)
         return child_node
 
     def is_terminal_node(self):
+        # get_done_reward() returns a tuple (done, reward)
+        # reward = ±1 when player 1 wins/loses the game
+        # reward = 0 when it is a tie
+        # reward = None when current node is not a teminal node
+        # done is the bool flag representing whether the game is over
         return self.env.get_done_reward()[0]
 
     def rollout(self):
+        """
+        Overview:
+            Traverse the tree starting from the current node until reaching a terminal node.
+            Take action according to the rollout policy, such as random policy.
```

Comment:
"""
    This method performs a rollout (simulation) from the current node.

    It repeatedly selects an action based on the rollout policy and simulates the action until the game ends.
    The method then returns the reward of the game's final state.
    """